queries.sql

```diff
@@ -106,4 +106,83 @@ SELECT COUNT (*) FROM animals JOIN owners ON animals.owners_id = owners.id WHERE
 SELECT COUNT (*) FROM animals JOIN owners ON animals.owners_id = owners.id WHERE owners.id = 5;
-SELECT COUNT (*) FROM animals JOIN owners ON animals.owners_id = owners.id WHERE owners.id = 6;
\ No newline at end of file
+SELECT COUNT (*) FROM animals JOIN owners ON animals.owners_id = owners.id WHERE owners.id = 6;
+
+/* Who was the last animal seen by William Tatcher? */
+
+SELECT animals.name 
+FROM animals 
+JOIN visits 
+ON animals.id = visits.animals_id 
+JOIN vets 
+ON vets.id = visits.vets_id 
+WHERE vets.id = 1 
+AND visits.visit_date = (
+    SELECT MAX(visit_date) 
+    FROM visits 
+    WHERE vets_id = 1
+);
+
+/* How many different animals did Stephanie Mendez see? */
+
+SELECT COUNT (*) 
+FROM animals 
+JOIN visits 
+ON animals.id = visits.animals_id 
+JOIN vets 
+ON vets.id = visits.vets_id 
+WHERE vets.id = 3;
+
+/* List all vets and their specialties, including vets with no specialties. */
+
+SELECT vets.name, species.name 
+FROM vets 
+FULL JOIN specializations 
+ON vets.id = specializations.vet_id 
+FULL JOIN species 
+ON species.id = specializations.species_id;
+
+/* List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020. */
+
+SELECT animals.name 
+FROM animals 
+JOIN visits 
+ON animals.id = visits.animals_id 
+JOIN vets 
+ON vets.id = visits.vets_id 
+WHERE vets.id = 3 
+AND visits.visit_date > '2020-05-01' 
+AND visits.visit_date < '2020-08-30';
+
+/* What animal has the most visits to vets? */
+
+SELECT COUNT(*) FROM animals JOIN visits ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE animals.id = 1;
+SELECT COUNT(*) FROM animals JOIN visits ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE animals.id = 2;
+SELECT COUNT(*) FROM animals JOIN visits ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE animals.id = 3;
+SELECT COUNT(*) FROM animals JOIN visits ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE animals.id = 4;
+SELECT COUNT(*) FROM animals JOIN visits ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE animals.id = 5;
+SELECT COUNT(*) FROM animals JOIN visits ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE animals.id = 6;
+SELECT COUNT(*) FROM animals JOIN visits ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE animals.id = 7;
+SELECT COUNT(*) FROM animals JOIN visits ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE animals.id = 8;
+SELECT COUNT(*) FROM animals JOIN visits ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE animals.id = 9;
+
+/* Details for most recent visit: */
+
+SELECT animals.name,vets.name,visit_date 
+FROM animals 
+JOIN visits 
+ON animals.id = visits.animals_id 
+JOIN vets 
+ON vets.id = visits.vets_id 
+WHERE  visits.visit_date = (
+    SELECT MIN(visit_date) 
+    FROM visits
+);
```

Comment:
- For this query, Kindly make sure to use `MAX(visit_date)` instead of `MIN(visit_date)` to get the most recent visit, not the oldest.